### DISCUSSION TITLE
Unable to load pimcore_enterprise_subscription_tools on Composer installation

### DISCUSSION THREAD

**Discussion by arulselvan008** [pimcore/pimcore#14748] [Q&A / Support] [answered]
as per the installation procedures

added in composee.json file 

    "repositories": {
        "pimcore_enterprise": {
            "type": "composer",
            "url": "https://enterprise.repo.pimcore.com/"
        }

 and config/config.yaml file:

pimcore_enterprise_subscription_tools:
    token: 'PIMCORE_ENTERPRISE_TOKEN'
    production_system: 'false'
    
Error Getting 
There is no extension able to load the configuration for "pimcore_enterprise_subscription_tools" (in "/var/www/pimcore/config/packages/dev/../../config.yaml"). Looked for namespace "pimcore_enterprise_subscription_tools", found ""framework", "security", "twig", "monolog", "doctrine", "doctrine_migrations", "sensio_framework_extra", "cmf_routing", "presta_sitemap", "scheb_two_factor", "fos_js_routing", "flysystem", "debug", "web_profiler", "pimcore", "pimcore_admin"" in /var/www/pimcore/config/packages/dev/../../config.yaml (which is being imported from "/var/www/pimcore/config/packages/dev/config.yaml").    

and try to add new bundles as needed for authentication.For this https://enterprise.repo.pimcore.com/, I don't have any credentials.

![image](https://user-images.githubusercontent.com/106297450/227498498-67497986-a0a7-4dea-a323-bb1010fba3af.png)

Does anybody know the solution?

**Comment by mattamon** [comment]
@arulselvan008 Is there a specific reason you want to install the enterprise bundles?
These bundles are only for enterprises and you need an enterprise subscription to get access to them. 

Please also check here: https://pimcore.com/editions

**Reply by arulselvan008** [reply]
Hi Mattamon,

We already bought the Enterprise edition, have a token, and are facing the issue during the installation process.

**Reply by fashxp** [reply, marked as answer]
it seems your token was not applied to composer. 
did you execute `composer config --global --auth http-basic.enterprise.repo.pimcore.com token <YOUR_TOKEN>` before starting composer require ...

**Reply by arulselvan008** [reply]
Hi Fashxp,

I created the skeleton package, added the subscription token, and made changes to the composer.json and config/config.yaml files as a starting point.

Elasticsearch and the portal engine installation were completed.

For your reference, the attached screenshot shows that we have completed bundles with the except the EnterpriseSubscriptionTool.

![image](https://user-images.githubusercontent.com/106297450/228242593-bd643aa5-431c-440e-8c53-bcdcebe003d5.png)

 Let me know, if there are any missing bundles need to be installed for enterprise installation.

**Reply by fashxp** [reply]
it depends on what features and extensions you need. 
the reason why the badge is orange and not green is because you have configured your system as development system. only production system are green. 
information about how to configure a system as a production system you find in the document where you received your token.

**Reply by arulselvan008** [reply]
Thanks  Fashxp,

We require PIM, DAM, and CMS perspectives for this, which we have installed below bundles.

![image](https://user-images.githubusercontent.com/106297450/228552616-84b962f1-5fff-4c02-9fee-9d1ad0e7358d.png)

For the above-mentioned features, do we need to install any extra bundles? Please suggest.

**Reply by fashxp** [reply]
no

**Comment by arulselvan008** [comment]
Thanks ,

I've finished setting up the portal engine, but I can't access the portal user login or portal page navigation. 

created a new portal setup wizard for geo-test created, and there are some pages, but I am unable to navigate the child document page in the PXP.

Document :-
![image](https://user-images.githubusercontent.com/106297450/229761197-4c663d8b-3f9b-46c2-bd92-fa9652410ffa.png)

Portal Page: 
![image](https://user-images.githubusercontent.com/106297450/229765013-9fff68a6-8e2c-4f57-a32f-625a38ce67eb.png)

Navigation & Properties:
![image](https://user-images.githubusercontent.com/106297450/229762556-e4d6eb65-f51d-4db6-b6aa-6b98bbc7e1d2.png)

while clicking the child document getting this error -
![image](https://user-images.githubusercontent.com/106297450/229885165-32482cca-c6d3-417d-8cad-d198875c7bf1.png)

If there are any configurations missing, please let me know.

**Comment by arulselvan008** [comment]
Hi ,

We attempted a new installation from scratch once more.

We purchased Pimcore's Enterprise version and attempted to install the Pimcore portal engine using the instructions provided below.
https://pimcore.com/docs/platform/Portal_Engine/
However, while installing the portal engine, we encountered the following error:

[ERROR] {"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST
         request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm="security"
         charset="UTF-8"","ApiKey"]}}],"type":"security_exception","reason":"missing authentication credentials for
         REST request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm="security"
         charset="UTF-8"","ApiKey"]}},"status":401}

We would appreciate your assistance in resolving this matter.
The following is our system configuration for installing the portal engine.

Pimcore version: 10.5.20
Elastic search: Elastic 8.7
Enterprise Metadata Bundle
Statistics Explorer

**Comment by arulselvan008** [comment]
Pimcore version: 10.5.20
Elastic search: Elastic 7.10 is suitable for the version
